src/main/java/net/consensys/orion/storage/QueryPrivacyGroupStorage.java

```diff
@@ -75,7 +75,7 @@ public String generateDigest(QueryPrivacyGroupPayload data) {
       if (result.isPresent()) {
         if (data.isToDelete()) {
           result.get().privacyGroupId().remove(data.privacyGroupToAppend());
-        } else {
+        } else if (!result.get().privacyGroupId().contains(data.privacyGroupToAppend())) {
```

Comment:
what happens if this condition is not met?

Reply:
If the condition isn't met then nothing happens to the ArrayList because it already contains the correct key - so to do nothing is the correct behavior.

Reply:
cool